api/src/main/java/javax/mvc/engine/ViewEngineContext.java

```diff
@@ -74,21 +71,6 @@
     Models getModels();
 
     /**
-     * Returns HTTP request object from the Servlet container.
-     *
-     * @return HTTP request object.
-     */
-    HttpServletRequest getRequest();
-
-    /**
-     * Returns HTTP response object from the servlet container. The underlying
-     * output stream should be used to write the result of processing a view.
-     *
-     * @return HTTP response object.
-     */
-    HttpServletResponse getResponse();
-
```

Comment:
Sorry, forget my last comment. Does not make too much sense. Without `@Produces` the matching would not work. That's the reason why Ozark is adding `text/html` [if the user did not](https://github.com/mvc-spec/ozark/blob/master/ozark/src/main/java/org/glassfish/ozark/jersey/OzarkModelProcessor.java#L85..L88).
The problem is that for the `ViewEngine` there's no difference if it's set by the user or the framework. 
I think your example would work if the encoding would be the only issue but what if a user wants to use `application/xhtml+xml`? 

As far as I've seen the `MediaType` is already populated in `ViewableWriter`. Ozark is reading the value from the headers which should be the same.